### DISCUSSION TITLE
How would you include (forced) convection cooling?

### DISCUSSION THREAD

**Discussion by TZ387** [ankrh/MCmatlab#10] [General]
Dear Anders and Dominik,

Thank you for your amazing product!

In near future, I intend to modify the MCmatlab code for my purposes, which is why I would like to hear your opinion on this mattter.

More precisely, I want to include (forced) convection cooling of tissue surface. Normally, such cooling (for example, with cold air) is included with the use of boundary condition on top of tissue, while the heat PDE for the "main body" $C_V\delta T/\delta t=q+\nabla \cdot(k\nabla T)$ remains the same:

$$k(\delta T(z=0,t)/\delta z) = h(T(z=0,t)-T_{air}).$$

Here, $k$ is the thermal conductivity on the surface of tissue, $h$ is the heat transfer coefficient [W m^-2 K^-1], and $T_{air}$ in this case represents temperature of cold air.

However, as such approach would give me more freedom, I am also interested in another possibility: Does there exists a physically correct way to somehow convert the written boundary condition to the "convective source term" in the heat PDE? 

More precisely, in that case, I would try to convert $C_V\delta T/\delta t=q+\nabla \cdot(k\nabla T)$ into something like this:

$$C_V\delta T/\delta t - \nabla \cdot(k\nabla T)=q+h(T_{conv}-T)/{\Delta}z,$$

where $h$ would be different from zero only for the voxels representing the top layer of the tissue. Also, $h(T_{conv}-T)/{\Delta}z$ could be generalized to $h\nabla T$, which would mean that a some form of Convection–diffusion equation (https://en.wikipedia.org/wiki/Convection%E2%80%93diffusion_equation) would be obtained. 

Still, I assume that such approach would be incorrect, which is why I would like to hear your opinion on this problem before I start implementing changes to the code.

Best regards, Tilen

**Comment by ankrh** [comment]
Hi Tilen

I think your idea is great. I'm not sure which way is most correct to write the actual equation, but I think your idea of using
$$C_V\delta T/\delta t - \nabla \cdot(k\nabla T)=q+h(T_{conv}-T)/{\Delta}z,$$
could probably work.

I think that one could also use an improvised hack in the media definitions of MCmatlab that could accomplish something similar. See the attached model file
[ConvectiveCooling.zip](https://github.com/ankrh/MCmatlab/files/11923281/ConvectiveCooling.zip)

Note that in this hack one would have to choose the (pseudo-) thermal conductivity of the air carefully to get the correct cooling rate on the tissue surface.

Let me know what you think.

**Comment by TZ387** [comment]
Hi Anders,

thank you for your answer! Due to other obligations, I am currently unable to provide more comprehensive answer, but on first glance, it does seem that such approach could work for at least some cases.

Looking at erratum of your mcMatlab article (https://doi.org/10.1117/1.JBO.26.1.019804), I assume that the $k_{'air'}$ (I included quotes to differentiate it from the real value) in the provided example should be obtained from the following relation:

$h=(2kk_{'air'}/(k+k_{'air'}))*(1/dz)$ ? In this equation, I of course neglected the real conductivity of air $k_{air}$, but I am currently unable to write the full equation due to LaTeX rendering issues.

**Reply by ankrh** [reply]
Sorry for the late reply! You have probably moved on by now but I'll provide my input for completeness and for anyone else who might be interested in this topic.

As far as I can tell your expression for h is correct. Certainly the effective thermal conductivity is calculated as in lines 460-466 in +MCmatlab\@model\simulateHeatDistribution.m, consistent with the equation in the erratum:
`%% Calculate proportionality between voxel-to-voxel temperature difference DeltaT and time step temperature change dT`
`TC_eff = 2*(TC*TC.')./(ones(nSM)*diag(TC)+diag(TC)*ones(nSM)); % Same as TC_eff(i,j) = 2*TC_red(i)*TC_red(j)/(TC_red(i)+TC_red(j)) but without for loops`
`TC_eff(isnan(TC_eff)) = 0; % Neighboring insulating voxels return NaN but should just be 0`

`dTdtperdeltaT  = cat(3,TC_eff./(diag(VHC)*ones(nSM))/model.G.dx^2,...`
`    TC_eff./(diag(VHC)*ones(nSM))/model.G.dy^2,...`
`    TC_eff./(diag(VHC)*ones(nSM))/model.G.dz^2); % [deg C /(s*deg C)] Time derivative of voxel temperature per voxel-to-voxel temperature difference. Third dimension corresponds to the different directions of heat diffusion (x, y and z)`
